Add support for moonlander (and other potential templates)

vladzaharia opened the pull request:
- Added support for multiple keyboard layouts based on geometry
- Added support for Moonlander Mk 1

Todo:
I'd like to add support for Planck's keyboard layout soon

vladzaharia commented on .gitignore:
```diff
@@ -0,0 +1,362 @@
+## Ignore Visual Studio temporary files, build results, and
```
This is just the standard C# gitignore from:

https://github.com/github/gitignore/blob/master/VisualStudio.gitignore

vladzaharia commented:
Moonlander;
![image](https://user-images.githubusercontent.com/79390/111896416-be153680-89d6-11eb-9f90-f2a661e69d00.png)

Planck:
![image](https://user-images.githubusercontent.com/79390/111896437-d9804180-89d6-11eb-9dec-84c409958e73.png)

Invvard commented:
First of all, thank you so much for this, seeing someone contributing is so motivating !
I'll review your PR tomorrow (AFK for the day) but as long as your code is tested and the existing code is working, that should go smoothly :)

vladzaharia commented:
It's an awesome tool, I just ordered my moonlander and thought it'd be perfect for it so happy to bring support for it. :)

Made sure to pass all the tests as well, of course:
![image](https://user-images.githubusercontent.com/79390/111913855-f0518300-8a2c-11eb-8885-47e6f12ae323.png)

vladzaharia commented on src/InvvardDev.EZLayoutDisplay.Tests/Service/LayoutServiceTest.cs:
```diff
@@ -14,27 +14,29 @@ public class LayoutServiceTest
     {
```
VS decided to get a little aggressive with the auto-formatting here, I'll revert it.

vladzaharia commented:
Fixed em all up and ready to review when you have the time @Invvard

Also added support for their new domain name for configurations (configure.zsa.io)

Invvard commented:
I've tried fixing the CI pipeline but failed... I've used legacy for quite sometime.
I'll fix it while putting it in YAML
